examples/sync/src/databases/variable.rs

```diff
@@ -0,0 +1,129 @@
+
+use crate::{Hasher, Key, Translator, Value};
+use commonware_cryptography::Hasher as CryptoHasher;
+use commonware_runtime::{buffer, Clock, Metrics, Storage};
+use commonware_storage::{
+    adb::{self, any::variable},
+    mmr::{hasher::Standard, verification::Proof},
+    store::operation,
+};
+use commonware_utils::{NZUsize, NZU64};
+use std::{future::Future, num::NonZeroU64};
+
+/// Variable-size Any database type alias.
+pub type Database<E> = variable::Any<E, Key, Value, Hasher, Translator>;
+
+/// Variable operation type alias.
+pub type Operation = operation::Variable<Key, Value>;
+
+/// Create a database configuration for use in tests.
+pub fn create_config() -> variable::Config<Translator, ()> {
+    variable::Config {
+        mmr_journal_partition: "mmr_journal".into(),
+        mmr_metadata_partition: "mmr_metadata".into(),
+        mmr_items_per_blob: NZU64!(4096),
+        mmr_write_buffer: NZUsize!(1024),
+        log_journal_partition: "log_journal".into(),
+        log_items_per_section: NZU64!(512),
+        log_compression: None,
+        log_codec_config: (),
+        log_write_buffer: NZUsize!(1024),
+        locations_journal_partition: "locations_journal".into(),
+        locations_items_per_blob: NZU64!(4096),
+        metadata_partition: "metadata".into(),
+        translator: Translator::default(),
+        thread_pool: None,
+        buffer_pool: buffer::PoolRef::new(NZUsize!(1024), NZUsize!(10)),
+    }
+}
+
+impl<E> crate::databases::Syncable for Database<E>
+where
+    E: Storage + Clock + Metrics,
+{
+    type Operation = Operation;
+
+    fn create_test_operations(count: usize, seed: u64) -> Vec<Self::Operation> {
+        let mut hasher = <Hasher as CryptoHasher>::new();
+        let mut operations = Vec::new();
+        for i in 0..count {
+            let key = {
+                hasher.update(&i.to_be_bytes());
+                hasher.update(&seed.to_be_bytes());
+                hasher.finalize()
+            };
+
+            let value = {
+                hasher.update(&key);
+                hasher.update(b"value");
+                hasher.finalize()
+            };
+
+            operations.push(Operation::Update(key, value));
+
+            if (i + 1) % 10 == 0 {
+                operations.push(Operation::Commit(None));
+            }
+        }
+
+        // Always end with a commit
+        operations.push(Operation::Commit(None));
+        operations
+    }
+
+    async fn add_operations(
+        database: &mut Self,
+        operations: Vec<Self::Operation>,
+    ) -> Result<(), commonware_storage::adb::Error> {
+        for operation in operations {
+            match operation {
+                Operation::Update(key, value) => {
+                    database.update(key, value).await?;
+                }
+                Operation::Delete(key) => {
+                    database.delete(key).await?;
+                }
+                Operation::Set(key, value) => {
```

Comment:
What is the difference here between `Set` and `Update`?

Reply:
There isn't -- `adb::variable` only has a notion of `update`, not `set`.

`create_test_operations` only returns `Update` and `Commit` operations, but the `Operation` type has other enum values that we need to handle. 

I updated the code to make it more clear that we only expect `Update` and `Commit` here in this test.